### ISSUE TITLE
successCallback can your add what's detected? : isRunningOnEmulator is FALSE POSITIVE

### ISSUE THREAD

**Issue by iPMisterX** [WuglyakBolgoink/cordova-plugin-iroot#42]
### My Apps use
cordova-plugin-iroot@2.0.2  
successCallback return **true**

> Vivo Y11 Android 9

but i check root with [Root Beer](https://play.google.com/store/apps/details?id=com.scottyab.rootbeer.sample)

![line_oa_chat_200826_132421](https://user-images.githubusercontent.com/11723453/91294118-3d38e800-e7c3-11ea-96cd-1ab2db4039fa.jpg)

### successCallback return true but i don't know what detected?
checkRootManagementApps
checkPotentiallyDangerousApps
checkRootCloakingApps
checkTestKeys
checkForDangerousProps
checkForBusyBoxBinary
checkForSuBinary
checkSuExists
checkForRWSystem
...or something is detected

**Comment by iPMisterX** [comment]
I/RootBeer: LOOKING FOR BINARY: /data/local/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /data/local/bin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /data/local/xbin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /sbin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /su/bin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/bin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/bin/.ext/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/bin/failsafe/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/sd/xbin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/usr/we-need-root/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/xbin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /cache/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /data/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /dev/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /system/sbin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /vendor/bin/su Absent :(
I/RootBeer: LOOKING FOR BINARY: /vendor/xbin/su Absent :(
E/IRoot: [canExecuteCommand] Error: Cannot run program "/system/xbin/which": error=2, No such file or directory
I/chromium: [INFO:CONSOLE(619)] **"isRooted: => true"**

**Comment by WuglyakBolgoink** [comment]
Hallo @iPMisterX I'll check this!

**Comment by iPMisterX** [comment]
# Temporary fix by comment this line 
`|| Build.BOARD.equals("QC_Reference_Phone") //bluestacks`

from this  plugins/cordova-plugin-iroot/src/android/de/cyberkatze/iroot/InternalRootDetection.java
```
public boolean isRunningOnEmulator() {
        Utils.getDeviceInfo();

        boolean simpleCheck = Build.MODEL.contains("Emulator")
            // ||Build.FINGERPRINT.startsWith("unknown") // Meizu Mx Pro will return unknown, so comment it!
            || Build.MODEL.contains("Android SDK built for x86")
            // || Build.BOARD.equals("QC_Reference_Phone") //bluestacks
            || Build.HOST.startsWith("Build"); //MSI App Player

        boolean checkGenymotion = Build.MANUFACTURER.contains("Genymotion");
        boolean checkGeneric = Build.FINGERPRINT.startsWith("generic") || (Build.BRAND.startsWith("generic") && Build.DEVICE.startsWith("generic"));
        boolean checkGoogleSDK = Build.MODEL.contains("google_sdk") || "google_sdk".equals(Build.PRODUCT);

        boolean result = simpleCheck || checkGenymotion || checkGeneric || checkGoogleSDK;

        LOG.d(
            Constants.LOG_TAG,
            String.format(
                "[isRunningOnEmulator] result [%s] => [simpleCheck:%s][checkGenymotion:%s][checkGeneric:%s][checkGoogleSDK:%s]",
                result,
                simpleCheck,
                checkGenymotion,
                checkGeneric,
                checkGoogleSDK
            )
        );

        return result;
    }
```
but i know it will not detect bluestacks by the way,  I think the actual problem is from Chinese Manufacturers

## The following devices as i known
- Vivo Y11
- Oppo
- Redmi 5A
- Redmi Note 5A

### i hope you will find the way to fix this or the new way to detect bluestacks

**Comment by WuglyakBolgoink** [comment]
Hallo @iPMisterX thank you for additional information!

**Comment by iPMisterX** [comment]
# New case in sony
![image](https://user-images.githubusercontent.com/11723453/91934300-3d7e3980-ed15-11ea-99bf-a48dae1b06f6.jpeg)
![image](https://user-images.githubusercontent.com/11723453/91934306-4111c080-ed15-11ea-8cdf-137f9c507dd1.jpeg)

**Comment by iPMisterX** [comment]
## isRunningOnEmulator Method is the problem.
it must be an option to check Root with this Method
**IRoot must be only check device has been Rooted by default.**

```
IRoot.isRooted
IRoot.isRootedWithBusyBox
IRoot.isRootedOrEmulator
```

**Comment by WuglyakBolgoink** [comment]
@iPMisterX  I will do a new plugin release in a few days

**Comment by WuglyakBolgoink** [comment]
merged in release v2.1.0
